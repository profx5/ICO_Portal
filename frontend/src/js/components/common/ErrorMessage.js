import React from 'react';
import PropTypes from 'prop-types';
import styled from "styled-components"
import {media} from 'js/utils/media';


const ErrorMessage = ({text}) => {
    return (
        <Message>{text}</Message>
    )
}


ErrorMessage.propTypes = {
    text: PropTypes.string
}

export default ErrorMessage;

const Message = styled.span`
    display: block;
    flex-basis: 100%;
    margin-top: 10px;
    font-size: 14px;
    color: rgb(242, 109, 109);
    white-space: nowrap;
    ${media.xs} {
        font-size: 12px;
        margin-top: 5px;
        white-space: unset;
    }
`;
