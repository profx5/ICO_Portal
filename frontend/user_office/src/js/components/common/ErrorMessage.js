import React from 'react'
import styled from "styled-components"
import {media} from 'js/utils/media';


const ErrorMessage = ({text}) => {
    return (
        <Message>{text}</Message>
    )
}


export default ErrorMessage;

const Message = styled.span`
    display: block;
    flex-basis: 100%;
    margin-top: 10px;
    font-size: 14px;
    color: rgb(242, 109, 109);
    white-space: nowrap;
    ${media.xs} {
        font-size: 12px;
        margin-top: 5px;
        white-space: unset;
    }
`;
