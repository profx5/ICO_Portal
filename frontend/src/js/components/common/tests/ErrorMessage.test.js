import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ErrorMessage from 'js/components/common/ErrorMessage';


configure({ adapter: new Adapter() });

it('should render stateless ErrorMessage', () => {
    const component = shallow(<ErrorMessage/>);
    expect(component).toMatchSnapshot();
})
