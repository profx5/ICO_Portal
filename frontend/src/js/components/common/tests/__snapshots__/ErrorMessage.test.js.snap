// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless ErrorMessage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: block;
    flex-basis: 100%;
    margin-top: 10px;
    font-size: 14px;
    color: rgb(242, 109, 109);
    white-space: nowrap;
    ",
          "@media (max-width: 767px)",
          " {
        font-size: 12px;
        margin-top: 5px;
        white-space: unset;
    }
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: block;
    flex-basis: 100%;
    margin-top: 10px;
    font-size: 14px;
    color: rgb(242, 109, 109);
    white-space: nowrap;
    ",
            "@media (max-width: 767px)",
            " {
        font-size: 12px;
        margin-top: 5px;
        white-space: unset;
    }
",
          ],
        },
        "displayName": "styled.span",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
